### DISCUSSION TITLE
aria-hidden behaviour doesn't play well with ProseMirror

### DISCUSSION THREAD

**Discussion by tslocke** [chakra-ui/zag#1745] [General]
Not sure if this counts as a bug or a feature request...

ProseMirror (rich text editor) has a feature called node-views that allow you to embed custom UI widgets inside the editable document. One thing you have to be careful of is that ProseMirror uses MutationObserver to keep its internal state in sync with contenteditable edits, and interactive node-views can wrongly trigger it which causes bugs. The node-view api includes a method `ignoreMutation` which let's you fix these bugs, but it's not sufficient for zag/combobox.

That's because the node-view can only ignore dom mutations that happen inside its root element, and the aria-hidden functionality you have makes changes outside of that element. These changes cause ProseMirror to rebuild the node-view. In other words you can't interact with it at all - as soon as you do something that would open the popup it all resets.

I've tried commenting out the body of this method in `combobox.machine`:
```ts
        hideOtherElements(ctx) {
          return ariaHidden([dom.getInputEl(ctx), dom.getContentEl(ctx), dom.getTriggerEl(ctx)])
        },
```
and it fixes the problem.

Is there any supported way to disable that feature? If not would you be willing to look at a PR?

I realise it's maybe bad from an accessibility point of view. Any suggestions for a better way to fix this issue?

Zag looks like incredible work - can't believe I only just discovered it!

**Comment by segunadebayo** [comment]
Interesting! Will consider adding a way to disable that behavior

**Reply by tslocke** [reply]
Any update on this? I'm still relying on patch-package in order to use zag in my app. Thanks.

**Reply by darkyelox** [reply]
I discovered that setting `aria-live` to an element prevents Zag from setting `aria-hidden` and `data-aria-hidden` attributes to that element, I saw that condition in the code of the `@zag-js/aria-hidden` package, I'm using version 1.3.3 for all Zag's packages.

**Reply by tslocke** [reply]
Thanks for the info. As I remember I was seeing those attributes get added to many, many elements. So I'm not sure that's going to solve it.
